### PR TITLE
Show file stats, nice sizes

### PR DESCRIPTION
**Pull request by akarve**

### CODE REVIEW

**Comment by akarve** [comment]
Resolves #427

File: catalog/app/containers/Package/index.js [hunk 1]
```diff
@@ -224,20 +225,49 @@ Install.propTypes = {
   owner: PropTypes.string.isRequired,
 };
 
+const sizes = [
```
Comment (dimaryaz) [inline catalog/app/containers/Package/index.js]: This feels overly complicated/verbose... Just do this? https://stackoverflow.com/questions/15900485/correct-way-to-convert-size-in-bytes-to-kb-mb-gb-in-javascript?answertab=active#tab-top

(Replace 1024 with 1000 or whatever.)

File: catalog/app/containers/Package/index.js [hunk 2]
```diff
@@ -224,20 +225,49 @@ Install.propTypes = {
...
+  { lim: 3, str: 'B' },
+  { lim: 6, str: 'kB' },
```
Comment (dimaryaz) [inline catalog/app/containers/Package/index.js]: Is `k` intentionally lowercase?

File: catalog/app/containers/Package/index.js [hunk 3]
```diff
@@ -224,20 +225,49 @@ Install.propTypes = {
...
+    display: inline-block;
+    width: 5em;
+  }
+`;
+function readableExtensions(fileCounts = {}) {
+  const keys = Object.keys(fileCounts);
+  keys.sort();
+  return keys.map((k) => {
+    const count = numberToCommaString(fileCounts[k]);
+    return <Line><span>.{k || 'None'}</span>{count}</Line>;
```
Comment (dimaryaz) [inline catalog/app/containers/Package/index.js]: Probably don't want `.None`... But actually, if I use `pathlib`, it'll give me extensions with leading dots - so I'll just return those to you.

File: catalog/app/containers/Package/index.js (hunk 2 above)
Comment (akarve) [reply catalog/app/containers/Package/index.js]: yeah i double-checked

File: catalog/app/containers/Package/index.js (hunk 1 above)
Comment (akarve) [reply catalog/app/containers/Package/index.js]: apparently `**` is faster than `Math.pow` (linter doesn't like `Math.pow` either). i think my solution should be the SO answer lol. seemed clean to me :) what we actually need are unit tests and @nl0 is working on those. unless this code is really bad would prefer to leave as is.

Comment (akarve) [reply catalog/app/containers/Package/index.js]: anyway, `**` is not really the issue. if you prefer i can take the log and jump right to the correct units if stepping through the array is hard to read.

Comment (dimaryaz) [reply catalog/app/containers/Package/index.js]: I didn't even mean `**` vs `pow` or `log`. I meant, the `lim: 3`, `lim: 6`, etc. are unnecessary - you can just use an array?

Comment (akarve) [reply catalog/app/containers/Package/index.js]: yeah, done

File: catalog/app/containers/Package/index.js [hunk 4]
```diff
@@ -224,20 +223,44 @@ Install.propTypes = {
...
-  const mb = bytes / 1000000;
-  if (mb < 1) {
-    const fracStr = (mb - Math.floor(mb)).toFixed(6);
-    // show fractional MB to 6 sig-digits
-    return `${fracStr} MB`;
+  if (Number.isInteger(bytes) && bytes > -1) {
+    // https://en.wikipedia.org/wiki/Kilobyte
+    const sizes = ['', 'k', 'M', 'G', 'T', 'P', 'E', 'Z', 'Y'];
+    const log = bytes === 0 ? 0 : Math.log10(bytes);
+    let index = Math.floor(log / 3);
```
[the remark this answers is not in the document]
Comment (akarve) [reply catalog/app/containers/Package/index.js]: yeah, `min` is nicer

**Pull request merged**